heartbeat/sender/peerAuthenticationSenderFactory.go

```diff
@@ -0,0 +1,129 @@
+	privKey                   crypto.PrivateKey
+	redundancyHandler         heartbeat.NodeRedundancyHandler
+}
+
+func newPeerAuthenticationSenderFactory(args argPeerAuthenticationSenderFactory) (*peerAuthenticationSenderFactory, error) {
+	return &peerAuthenticationSenderFactory{
+		messenger:                 args.messenger,
+		marshaller:                args.marshaller,
+		topic:                     args.topic,
+		timeBetweenSends:          args.timeBetweenSends,
+		timeBetweenSendsWhenError: args.timeBetweenSendsWhenError,
+		thresholdBetweenSends:     args.thresholdBetweenSends,
+		nodesCoordinator:          args.nodesCoordinator,
+		peerSignatureHandler:      args.peerSignatureHandler,
+		hardforkTrigger:           args.hardforkTrigger,
+		hardforkTimeBetweenSends:  args.hardforkTimeBetweenSends,
+		hardforkTriggerPubKey:     args.hardforkTriggerPubKey,
+		keysHolder:                args.keysHolder,
+		timeBetweenChecks:         args.timeBetweenChecks,
+		shardCoordinator:          args.shardCoordinator,
+		privKey:                   args.privKey,
+		redundancyHandler:         args.redundancyHandler,
+	}, nil
+}
+
+func (pasf *peerAuthenticationSenderFactory) create() (peerAuthenticationHandler, error) {
+	isMultikey := check.IfNil(pasf.privKey)
+	isRegular := check.IfNil(pasf.keysHolder)
+	if isMultikey && isRegular {
+		return nil, fmt.Errorf("%w for peerAuthenticationSenderFactory, received both validator key and multikey holder", heartbeat.ErrInvalidConfiguration)
+	}
+
+	if !isMultikey && !isRegular {
+		return nil, fmt.Errorf("%w for peerAuthenticationSenderFactory, could not determine node's type", heartbeat.ErrInvalidConfiguration)
+	}
+
+	if isMultikey {
+		return pasf.createMultikeyPeerAuthenticationSender()
+	}
+
+	return pasf.createPeerAuthenticationSender()
+}
+
+func (pasf *peerAuthenticationSenderFactory) createPeerAuthenticationSender() (*peerAuthenticationSender, error) {
+	args := argPeerAuthenticationSender{
+		argBaseSender: argBaseSender{
+			messenger:                 pasf.messenger,
+			marshaller:                pasf.marshaller,
+			topic:                     pasf.topic,
+			timeBetweenSends:          pasf.timeBetweenSends,
+			timeBetweenSendsWhenError: pasf.timeBetweenSendsWhenError,
+			thresholdBetweenSends:     pasf.thresholdBetweenSends,
+		},
+		nodesCoordinator:         pasf.nodesCoordinator,
+		peerSignatureHandler:     pasf.peerSignatureHandler,
+		privKey:                  pasf.privKey,
+		redundancyHandler:        pasf.redundancyHandler,
+		hardforkTrigger:          pasf.hardforkTrigger,
+		hardforkTimeBetweenSends: pasf.hardforkTimeBetweenSends,
+		hardforkTriggerPubKey:    pasf.hardforkTriggerPubKey,
+	}
+
+	return newPeerAuthenticationSender(args)
+}
+
+func (pasf *peerAuthenticationSenderFactory) createMultikeyPeerAuthenticationSender() (*multikeyPeerAuthenticationSender, error) {
+	args := argMultikeyPeerAuthenticationSender{
+		argBaseSender: argBaseSender{
```

Comment:
could have extracted a function `createBaseSenderArgs` and use it in the both functions

Reply:
not needed anymore with latest push